### PR TITLE
Fix #21407

### LINKED ISSUE #21407: VisualViewport in std/dom
### Summary

`std/dom` does not provide access to the `VisualViewport` API.

### Description

Adding `VisualViewport` to `std/dom` is trivial. The API is very simple.

`VisualViewport` provides information about the visual viewport of the window and iframes; in simpler terms, it helps your JS code understand what part of the window you're zoomed into. `VisualViewport` has been supported by all major browsers for about 1.5 years.

### Alternatives

I could create a Nimble package that provides extensions to std/dom, including implementations of things like `VisualViewport` and `IndexedDB`. Seems like a kludge, though.

### Examples

This code implements the VisualViewport API:

```nim
import std/dom

type
  VisualViewport* {.importc.} = ref object of EventTarget
    offsetLeft*: float
    offsetTop*: float
    pageLeft*: float
    pageTop*: float
    width*: float
    height*: float
    scale*: float
    onresize*: proc (event: Event) {.closure.}
    onscroll*: proc (event: Event) {.closure.}

proc visualViewport*(w: Window): VisualViewport {.importcpp: "#.visualViewport", nodecl.}
```

For use cases of VisualViewport, refer to the MDN docs linked below.

### Backwards Compatibility

Users that use their own VisualViewport implementation (like the one I shared above) may have some issues with conflicting typedefs/names.

### Links

[MDN description of VisualViewport](https://developer.mozilla.org/en-US/docs/Web/API/VisualViewport)

[Can I use visualViewport?](https://caniuse.com/#search=visualViewport)

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Fix #21407

### CODE REVIEW

File: lib/std/jsviewport.nim [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## - Viewport API for the JavaScript targets: https://developer.mozilla.org/en-US/docs/Web/API/VisualViewport
+when not defined(js):
+  {.fatal: "Module is designed to be used with the JavaScript backend.".}
+
+import std/dom
+
+type VisualViewport* {.importc.} = ref object of EventTarget
+  offsetLeft*, offsetTop*, pageLeft*, pageTop*, width*, height*, scale*: float
+  onResize*: proc (event: Event) {.closure.}
+  onScroll*: proc (event: Event) {.closure.}
```
Comment (AmjadHD) [inline lib/std/jsviewport.nim]:
```suggestion
  onResize*, onScroll*: proc (event: Event) {.closure.}
```

**Comment by Araq** [comment]
Why not add it to the DOM module?
